File: CPP/Program/HybridStorage.cpp
#pragma once
#include <iostream>
#include <string>
#include "Harddrive.cpp"
#include "Ssd.cpp"

using namespace std;

class HybridStorage : public Harddrive, public Ssd
{
private:
  int kapasitasGB;  // → Kapasitas penyimpanan
  string tipeDrive; // → "Hybrid"
  int rpm;          // → Jika ada rotasi dari HDD
  string nandType;  // → Jenis NAND dari SSD

public:
  HybridStorage()
  {
  }

  HybridStorage(int kapasitasGB, string tipeDrive, int rpm, string nandType,
                int kapasitasGB2, string tipeDrive2, string merk, string nama,
                int kapasitasGB3, string tipeDrive3, string merk2, string nama2) : Harddrive(kapasitasGB2, tipeDrive2, merk, nama), Ssd(kapasitasGB3, tipeDrive3, merk, nama)
  {
    this->kapasitasGB = kapasitasGB;
    this->tipeDrive = tipeDrive;
    this->rpm = rpm;
    this->nandType = nandType;
  }

  void setKapasitasGB(int kapasitasGB)
  {
    this->kapasitasGB = kapasitasGB;
  }

  void setTipeDrive(string tipeDrive)
  {
    this->tipeDrive = tipeDrive;
  }

  void setRpm(int rpm)
  {
    this->rpm = rpm;
  }

  void setNandType(string nandType)
  {
    this->nandType = nandType;
  }

  int getKapasitasGB()
  {
    return this->kapasitasGB;
  }

  string getTipeDrive()
  {
    return this->tipeDrive;
  }

  int getRpm()
  {
    return this->rpm;
  }

  string getNandType()
  {
    return this->nandType;
  }

  int getTotalKapasitasGB()
  {
    return Harddrive::getKapasitasGB() + Ssd::getKapasitasGB();
  }

  // Getter untuk mengambil informasi SSD dan HDD
  Ssd getSsd()
  {
    return *this; // Menggunakan pewarisan untuk mengembalikan Ssd
  }

  Harddrive getHarddrive()
  {
    return *this; // Menggunakan pewarisan untuk mengembalikan Harddrive
  }

  // Getter untuk mendapatkan merk dari HDD
  string getHddMerk()
  {
    return Harddrive::getMerk(); // Ambil dari Harddrive
  }

  // Getter untuk mendapatkan nama dari HDD
  string getHddNama()
  {
    return Harddrive::getNama(); // Ambil dari Harddrive
  }

  // Getter untuk mendapatkan merk dari SSD
  string getSsdMerk()
  {
    return Ssd::getMerk(); // Ambil dari Ssd
  }

  // Getter untuk mendapatkan nama dari SSD
  string getSsdNama()
  {
    return Ssd::getNama(); // Ambil dari Ssd
  }

  ~HybridStorage()
  {
  }
};

File: JAVA/Program/HybridStorage.java
public class HybridStorage {
  private int kapasitasGB;  // → Kapasitas penyimpanan
  private String tipeDrive; // → "Hybrid"
  private int rpm;          // → Jika ada rotasi dari HDD
  private String nandType;  // → Jenis NAND dari SSD
  private Harddrive harddrive;
  private Ssd ssd;

  public HybridStorage() {
  }

  public HybridStorage(int kapasitasGB, String tipeDrive, int rpm, String nandType,
             int kapasitasGB2, String tipeDrive2, String merk, String nama,
             int kapasitasGB3, String tipeDrive3, String merk2, String nama2) {
    this.kapasitasGB = kapasitasGB;
    this.tipeDrive = tipeDrive;
    this.rpm = rpm;
    this.nandType = nandType;
    this.harddrive = new Harddrive(kapasitasGB2, tipeDrive2, merk, nama);
    this.ssd = new Ssd(kapasitasGB3, tipeDrive3, merk2, nama2);
  }

  public void setKapasitasGB(int kapasitasGB) {
    this.kapasitasGB = kapasitasGB;
  }

  public void setTipeDrive(String tipeDrive) {
    this.tipeDrive = tipeDrive;
  }

  public void setRpm(int rpm) {
    this.rpm = rpm;
  }

  public void setNandType(String nandType) {
    this.nandType = nandType;
  }

  public int getKapasitasGB() {
    return this.kapasitasGB;
  }

  public String getTipeDrive() {
    return this.tipeDrive;
  }

  public int getRpm() {
    return this.rpm;
  }

  public String getNandType() {
    return this.nandType;
  }

  public int getTotalKapasitasGB() {
    return this.harddrive.getKapasitasGB() + this.ssd.getKapasitasGB();
  }

  // Getter untuk mengambil informasi SSD dan HDD
  public Ssd getSsd() {
    return this.ssd;
  }

  public Harddrive getHarddrive() {
    return this.harddrive;
  }

  // Getter untuk mendapatkan merk dari HDD
  public String getHddMerk() {
    return this.harddrive.getMerk();
  }

  // Getter untuk mendapatkan nama dari HDD
  public String getHddNama() {
    return this.harddrive.getNama();
  }

  // Getter untuk mendapatkan merk dari SSD
  public String getSsdMerk() {
    return this.ssd.getMerk();
  }

  // Getter untuk mendapatkan nama dari SSD
  public String getSsdNama() {
    return this.ssd.getNama();
  }
}
